src/streamlink/plugins/pandalive.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin, PluginError, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://(?:www\.)?pandalive\.co\.kr/"
+))
+class Pandalive(Plugin):
+    API_URL = "https://api.pandalive.co.kr/v1/live/play"
+
+    _room_id_re = re.compile(r"roomid\s*=\s*String\.fromCharCode\((.*)\)")
+
+    def _get_media_code(self):
+        try:
+            return self.session.http.get(self.url, schema=validate.Schema(
+                validate.parse_html(),
+                validate.xml_xpath_string(".//script[contains(text(), 'roomid')]/text()"),
+                str,
+                validate.transform(self._room_id_re.search),
+                validate.any(None, validate.all(
+                    validate.get(1),
+                    validate.transform(lambda s: "".join(map(lambda c: chr(int(c)), s.split(",")))),
+                )),
+            ))
+        except PluginError:
+            pass
+
+        return None
+
+    def _get_streams(self):
+        media_code = self._get_media_code()
+
+        if not media_code:
+            return
+
+        log.debug(f"Media code: {media_code}")
+
+        data = {
+            "action": "watch",
+            "mediaCode": media_code,
+        }
+        json = self.session.http.post(self.API_URL, data=data, schema=validate.Schema(
```

Comment:
```py
        data = self.session.http.post(
            "https://api.pandalive.co.kr/v1/live/play",
            data={"action": "watch", "mediaCode": media_code},
            schema=validate.Schema(
                validate.parse_json(),
```

Reply:
Is it now preferred to embed the API URLs like this, verses defining them within the class?  Perhaps you're suggesting this because in this case it's only used once anyway?